### PR TITLE
REF: implement should_extension_dispatch

### PR DESCRIPTION
**Pull request by jbrockmendel**
This uses should_extension_dispatch in _arith_method_SERIES.  We'll be able to use it in _comp_method_SERIES after #27803. (this is orthogonal to 27803 so doesn't matter what order they go in)

### CODE REVIEW

File: pandas/core/ops/__init__.py [hunk 1]
```diff
@@ -460,6 +460,33 @@ def masked_arith_op(x, y, op):
 # Dispatch logic
 
 
+def should_extension_dispatch(left, right) -> bool:
```
Comment (jreback) [inline pandas/core/ops/__init__.py]: can you type left & right

**Review by jreback — COMMENTED** [review]
minor comment otherwise lgtm.

**Comment by jbrockmendel** [comment]
Updated and green

**Comment by jbrockmendel** [comment]
@jreback gentle ping; this is a step toward the Plan B discussed on the call for arithmetic perf

**Comment by jreback** [comment]
thanks

**Pull request merged**